fastestimator/trace/io/model_eval.py

```diff
@@ -0,0 +1,451 @@
+
+            with self.doc.create(Table(position='H')) as table:
+                table.append(NoEscape(r'\refstepcounter{table}'))
+                self._document_summary_table(pass_num=len(instance_pass_tests) + len(aggregate_pass_tests),
+                                             fail_num=len(instance_fail_tests) + len(aggregate_fail_tests))
+
+        if instance_fail_tests or aggregate_fail_tests:
+            with self.doc.create(Section("Failed Tests")):
+                if len(aggregate_fail_tests) > 0:
+                    with self.doc.create(Subsection("Failed Aggregate Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_aggregate_table(tests=aggregate_fail_tests)
+                if len(instance_fail_tests) > 0:
+                    with self.doc.create(Subsection("Failed Per-Instance Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_instance_table(tests=instance_fail_tests, with_ID=self.data_id)
+
+        if instance_pass_tests or aggregate_pass_tests:
+            with self.doc.create(Section("Passed Tests")):
+                if aggregate_pass_tests:
+                    with self.doc.create(Subsection("Passed Aggregate Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_aggregate_table(tests=aggregate_pass_tests)
+                if instance_pass_tests:
+                    with self.doc.create(Subsection("Passed Per-Instance Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_instance_table(tests=instance_pass_tests, with_ID=self.data_id)
+
+    def _document_summary_table(self, pass_num: int, fail_num: int) -> None:
+        """Document a summary table.
+
+        Args:
+            pass_num: Total number of passed tests.
+            fail_num: Total number of failed tests.
+        """
+        with self.doc.create(Tabularx('|Y|Y|Y|', booktabs=True)) as tabular:
+            package = Package('seqsplit')
+            if package not in tabular.packages:
+                tabular.packages.append(package)
+
+            # add table heading
+            tabular.add_row(("Total test #", "Total pass #", "Total fail #"), strict=False)
+            tabular.add_hline()
+
+            tabular.add_row((pass_num + fail_num, pass_num, fail_num), strict=False)
+
+    def _document_instance_table(self, tests: List[dict], with_ID: bool):
```

Comment:
Can you put typing on the dict here? List[Dict[str, TestCase]] or w/e it is

Reply:
The tests here is a list of dict that maps string to (string, int, bool).
The dict has some keys that match TestCase attribute but it is not TestCase.